src/Ledger/Utxow.lagda

```diff
@@ -152,29 +152,38 @@ data _⊢_⇀⦇_,UTXOW⦈_ where
 \end{code}
 \begin{code}
   UTXOW-inductive :
-    let open Tx tx renaming (body to txb); open TxBody txb; open TxWitnesses wits
-        open UTxOState s
-        witsKeyHashes     = mapˢ hash (dom vkSigs)
-        witsScriptHashes  = mapˢ hash scripts
-        inputHashes       = getInputHashes tx utxo
-        refScriptHashes   = fromList $ map hash (refScripts tx utxo)
-        neededHashes      = scriptsNeeded utxo txb
-        txdatsHashes      = dom txdats
-        allOutHashes      = getDataHashes (range txouts)
-        nativeScripts     = mapPartial isInj₁ (txscripts tx utxo)
+\end{code}
+\begin{code}[hide]
+    let  open Tx tx renaming (body to txb); open TxBody txb; open TxWitnesses wits
+         open UTxOState s
```

Comment:
In line with what I said on the issue about formatting, this could instead be a `⟦ utxo , ... , ... ⟧ = s` and moved to the visible section. For the `Tx` stuff I think we should definitely have some general explanation like with `PParams`.

Reply:
I don't know what formatting issue you're referring to, so I can make sense of the suggestion `⟦ utxo , ... , ... ⟧ = s`. Please provide a link or more specific reference.

Reply:
Perhaps you're suggesting I remove the `open UTxOState s` and then, in visible code, have

```agda
    ∙  Γ ⊢ ⟦ utxo , fees , deps , dons ⟧ᵘ ⇀⦇ tx ,UTXO⦈ ⟦ utxo' , fees' , deps' , dons' ⟧ᵘ
       ────────────────────────────────
       Γ ⊢ ⟦ utxo , fees , deps , dons ⟧ᵘ ⇀⦇ tx ,UTXOW⦈ ⟦ utxo' , fees' , deps' , dons' ⟧ᵘ
```

where `utxo`, `fees`, `deps`, `dons` (and their primed counterparts) are all private variables.  Please confirm.

Reply:
...or, rather, put `⟦ utxo , fees , deps , dons ⟧ᵘ = s` in the visible `let` code?

Reply:
Okay, that works... I now have the following in the visible `let` block:
```agda
        ⟦ utxo , fees , deps , dons ⟧ᵘ      = s
        ⟦ utxo' , fees' , deps' , dons' ⟧ᵘ  = s'
```

Reply:
I think in the `let` it has the advantage of not needing to spell out every variable twice. But either way looks fine to me.